### DISCUSSION TITLE
Using foreign keys in filtering?

### DISCUSSION THREAD

**Discussion by Atlinx** [twinlogix/typetta#204] [Q&A] [answered]
Is it possible to filter for the specific elements of a foreign reference?

I'd like to do something like this
```typescript
// Project has projectMembers as a foreignref
// ProjectMember has roles as a foreignref
const projects = entityManager.projects.findAll({
  filter: {
    projectMembers: { containsElement: {
        userId: { eq: "someUserId" },
        roles: { containsElement: {
          roleCode: { eq: "PROJECT_OWNER" }
        }}
      }
    },
  }
});
```
however, a `containsElement` operation doesn't exist.

**Comment by edobrb** [comment, marked as answer]
If you want to filter `Project` entities based on the `projectMembers` entity it is not yet possible, but it is in the roadmap (see issue #13). To work around this problem you can simply query the `ProjectMembers` entities and then search for the `Project` that are related to the result of the previous query.

Instead, if you want to filter the foreignRef or innerRef fields you can use `relations` field inside the find operations.
